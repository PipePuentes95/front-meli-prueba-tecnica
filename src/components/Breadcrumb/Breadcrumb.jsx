import React from 'react'
import './Breadcrumb.scss'

const Breadcrumb = ({ categories = [] }) => {

  return (
    <div className='breadcrumb'>
      <div className="breadcrumb-content">
        { categories.map((category) => (
          <span className='breadcrumb-item' key={ category }>{ category }</span>
        )) }
      </div>
    </div>
  );
};

export default Breadcrumb;
