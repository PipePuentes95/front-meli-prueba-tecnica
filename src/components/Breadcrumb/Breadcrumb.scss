@import '../../styles/variables';

.breadcrumb {
  @media(max-width: 764px) {
    width: 100%;
  }

  width: 80%;

  margin: 0 auto;
  padding: 16px 0;

  .breadcrumb-content {
    display: flex;
    flex-wrap: wrap;

    @media(max-width: 576px) {
      margin-bottom: -5px;
    }

    .breadcrumb-item {

      @media(max-width: 576px) {
        padding-bottom: 5px;
      }

      font-size: 14px;
      color: $grey-3;

      &:not(:last-of-type):after {
        margin: 0px 8px;

        font-size: 22px;
        line-height: 0;
    
        content: 'â€º';
      }
    }
  }

}
