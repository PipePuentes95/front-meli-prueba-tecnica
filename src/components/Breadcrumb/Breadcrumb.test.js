import { render } from '@testing-library/react';
import Breadcrumb from './Breadcrumb';

const categories = ["audio", "tecnologia", "celular"]

it('should take a snapshot', () => {
  const { asFragment } = render(<Breadcrumb categories={ categories } />);

  expect(asFragment(<Breadcrumb />)).toMatchSnapshot();
});
