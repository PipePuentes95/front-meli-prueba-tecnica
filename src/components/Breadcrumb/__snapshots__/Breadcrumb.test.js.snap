// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb"
  >
    <div
      class="breadcrumb-content"
    >
      <span
        class="breadcrumb-item"
      >
        audio
      </span>
      <span
        class="breadcrumb-item"
      >
        tecnologia
      </span>
      <span
        class="breadcrumb-item"
      >
        celular
      </span>
    </div>
  </div>
</DocumentFragment>
`;
